### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by dheerajsachdeva**
- Fixed n+1 problems for Posts of User

- Before Screenshot

screenshot11

After screenshot (Can See arrays while fetching posts for user):

image

- Implemented Capybara Tests for the followings:

**User index page:**

- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.

**user show page:**

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.

**User post index page:**

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.

**Post show page:**

- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left.

😄

### CODE REVIEW

File: spec/user_show_page_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    it 'should show last three posts only' do
+      expect(page).to have_content(@fourth_post.title)
+      expect(page).to have_content(@third_post.title)
+      expect(page).to have_content(@second_post.title)
+    end
+
+    it 'should have a link to see all posts' do
+      expect(page.html).to include('See All Posts')
+    end
```
Comment (Davidosky007) [inline spec/user_show_page_spec.rb:44]: * Kindly write tests for these cases as it's missing `When I click a user's post, it redirects me to that post's show page.`

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @dheerajsachdeva @Ade179 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* N +1 issue solved 👍 
*  Integration tested with Capybara gem properly 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/user_show_page_spec.rb (hunk 1 above)
Comment (dheerajsachdeva) [reply spec/user_show_page_spec.rb:44]: @Davidosky007 Hello,

- Test added. Thanks 😄

**Review by prolajumokeoni — APPROVED** [review]
Hi @dheerajsachdeva 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All changes requested have been implemented 🥇    

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**